### ISSUE TITLE
Complement Postgres & Complement workers: de-flaking

### ISSUE THREAD

**Issue by reivilibre** [matrix-org/synapse#13161]
The Postgres and workers Complement runs, which have recently been enabled, are a bit flakey at times.

Let's track the list of flakey tests here and get them cleaned up.

- [ ] TestFederationRoomsInvite/Parallel/Invited_user_can_reject_invite_over_federation_for_empty_room
- [x] TestRestrictedRoomsLocalJoinInMSC3787Room/Join_should_fail_when_left_allowed_room and TestRestrictedRoomsLocalJoin}/Join_should_fail_when_left_allowed_room: https://github.com/matrix-org/synapse/runs/7179299458?check_suite_focus=true
- [x] TestRestrictedRoomsLocalJoin/Join_should_succeed_when_joined_to_allowed_room and TestRestrictedRoomsLocalJoinInMSC3787Room/Join_should_succeed_when_joined_to_allowed_room: https://github.com/matrix-org/synapse/runs/7179299458?check_suite_focus=true
   - #13185
   - https://github.com/matrix-org/complement/pull/413

Once all this is done, we can close #13264.

**Comment by squahtx** [comment]
TestRestrictedRoomsLocalJoinInMSC3787Room: https://github.com/matrix-org/synapse/runs/7156318189?check_suite_focus=true

**Comment by DMRobertson** [comment]
*  TestMembersLocal/Parallel/Existing_members_see_new_members'_presence

https://github.com/matrix-org/synapse/runs/7184655709?check_suite_focus=true

**Comment by DMRobertson** [comment]
*  TestWriteMDirectAccountData https://github.com/matrix-org/synapse/runs/7184887268?check_suite_focus=true

**Comment by squahtx** [comment]
* TestSync/parallel/sync_should_succeed_even_if_the_sync_token_points_to_a_redaction_of_an_unknown_event
  https://github.com/matrix-org/synapse/runs/7197178106?check_suite_focus=true#step:4:6887

**Comment by DMRobertson** [comment]
- [ ] TestRestrictedRoomsLocalJoinInMSC3787Room/Join_should_succeed_when_invited
    - caused by #13269

**Comment by DMRobertson** [comment]
[ ]  TestRoomState/Parallel/GET_/publicRooms_lists_newly-created_room
- https://github.com/matrix-org/synapse/runs/7337130834?check_suite_focus=true

**Comment by DMRobertson** [comment]
- [ ] TestRestrictedRoomsRemoteJoinFailOverInMSC3787Room
  - https://github.com/matrix-org/synapse/runs/7397381295?check_suite_focus=true (postgres monolith)

**Comment by reivilibre** [comment]
- [ ] TestImportHistoricalMessages/parallel/Federation/When_messages_have_already_been_scrolled_back_through,_new_historical_messages_are_visible_in_next_scroll_back_on_federated_server https://github.com/matrix-org/synapse/runs/7609443787?check_suite_focus=true
- [ ]

**Comment by reivilibre** [comment]
Superseded by:
- https://github.com/matrix-org/synapse/issues/13508
- https://github.com/matrix-org/synapse/issues/13173#issuecomment-1190444175
- https://github.com/matrix-org/synapse/issues/13334
